secant/Features/App/AppStore.swift

```diff
@@ -192,13 +193,18 @@ extension AppReducer {
                     with: environment
                 )
 
-                try environment.SDKSynchronizer.prepareWith(initializer: initializer, seedBytes: seedBytes)
-                try environment.SDKSynchronizer.start()
+                return Effect.task {
+                    try await environment.SDKSynchronizer.prepareWith(initializer: initializer, seedBytes: seedBytes)
+                    try environment.SDKSynchronizer.start()
+                    return .sdkInitialized
+                }
             } catch {
-                print("Prepare error: \(error)")
                 state.appInitializationState = .failed
                 // TODO [#221]: error we need to handle (https://github.com/zcash/secant-ios-wallet/issues/221)
+                return .none
             }
+
+        case .sdkInitialized:
             return .none
```

Comment:
Looks like `sdkInitialized` is not used in the code. I would assume you just need to run the async code and forget it. For it you can use 
```swift
return Effect.fireAndForget {
    (try) await someAsync()
}
```

Reply:
I tried to do it in that way. But after all I can't. I need to catch error inside Effect.task otherwise that error won't be catched. And I need a way how to react on that error. And that way is to introduce new actions (`sdkInitialized` and `sdkInitializationFailed`) which then we can handle however we want. In theory I would be fine with `sdkInitializationFailed` but Effect must always return any action so I need `sdkInitialized` also.

It's very similar case as in second comment. I just forgot about do catch here.

Reply:
@LukasKorba Can you please check my answer? Thanks.